src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,54 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_STORAGE_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int index = 0;
+
+    public StorageImpl() {
+        try {
+            keys = (K[]) new Object[DEFAULT_STORAGE_SIZE];
+            values = (V[]) new Object[DEFAULT_STORAGE_SIZE];
+        } catch (Exception e) {
+            throw new RuntimeException("Something went wrong...", e);
+        }
+
+    }
+
     @Override
     public void put(K key, V value) {
+        if (existingKeySearch(key) >= 0) {
+            values[existingKeySearch(key)] = value;
+            return;
+        }
+        keys[index] = key;
+        values[index] = value;
+        index++;
+    }
+
+    private int existingKeySearch(K key) {
```

Comment:
put private methods below public
```suggestion
    private int searchKey(K key) {
```